docs/index/docs/typedoc-hookstate-core.md

```diff
@@ -699,6 +699,19 @@ ___
 
 *Defined in [index.d.ts:296](https://github.com/avkonst/hookstate/blob/master/dist/index.d.ts#L296)*
 
+**A note about previous values and merging:** State values are muteable in Hookstate for performance reasons. This causes a side effect in the merge operation. While merging, the previous state object is mutated as the desired changes are applied. This renders the previous property unreliable as it will match the new state property. While the [merged](#optional-readonly-merged) property can be used to detect which values were merged in, it will not inform you whether those values are different from the previous state.
```

Comment:
Hi, thanks for the note.
This file is automatically generated from documentation comments in source code. So it should be added to the source code instead.

Reply:
"This renders the previous property unreliable as it will match the new state property." maybe it is me, but I think this sentence is unclear.

Reply:
I've moved the note to the type definition and reworded for clarity. Also ran build and build:docs.